src/main/java/net/snowflake/client/core/HttpUtil.java

```diff
@@ -69,16 +69,51 @@
 
   public static final String JDBC_TTL = "net.snowflake.jdbc.ttl";
 
-  /** The unique httpClient shared by all connections. This will benefit long- lived clients */
-  private static Map<OCSPMode, CloseableHttpClient> httpClient = new ConcurrentHashMap<>();
+  /**
+   * Object to provide the proxy user and password credentials for the HTTP Client object when
+   * needed
+   */
+  private static CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
 
   /**
-   * The unique httpClient shared by all connections that don't want decompression. This will
-   * benefit long-lived clients
+   * The unique httpClient shared by all connections. This will benefit long- lived clients. Key =
+   * proxy host + proxy port + nonProxyHosts, Value = Map<OCSPMode, HttpClient>
    */
-  private static Map<OCSPMode, CloseableHttpClient> httpClientWithoutDecompression =
+  private static Map<String, Map<OCSPMode, CloseableHttpClient>> httpClientWithProxies =
       new ConcurrentHashMap<>();
 
+  /**
+   * The unique httpClient map shared by all connections that don't want decompression. This will
+   * benefit long-lived clients. Key = proxy host + proxy port + nonProxyHosts, Value =
+   * Map<OCSPMode, HttpClient>
+   */
+  private static Map<String, Map<OCSPMode, CloseableHttpClient>>
+      httpClientWithoutDecompressionWithProxies = new ConcurrentHashMap<>();
+
+  /**
+   * Creates a string value that is used as a key for the httpClient hashmaps so that a new
+   * httpClient is created for each new proxy configuration
+   *
+   * @return string used as a key to the hashmap
+   */
+  private static String computeProxySettings() {
+    return useProxy ? proxyHost + proxyPort + nonProxyHosts : "false";
+  }
+
+  /**
+   * Helper function to count the number of HttpClients. For testing purposes only.
+   *
+   * @return
+   */
+  public static int countTotalHttpClientsInMap() {
```

Comment:
For testing only- making sure that httpclient objects are getting reused in the map